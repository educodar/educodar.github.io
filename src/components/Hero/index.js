import { GatsbyLinkButton, H2, Paragraph } from "muy"
import React from "react"
import makeStyles from "@material-ui/core/styles/makeStyles"
import { Card } from "@material-ui/core"
import CardMedia from "@material-ui/core/CardMedia"
import CardContent from "@material-ui/core/CardContent"
import CardActions from "@material-ui/core/CardActions"

const useStyles = makeStyles((theme) => ({
  h2: {
    color: theme.palette.primary.dark,
    fontSize: "2rem",
    fontWeight: "bold",
  },
}))

const Hero = ({ heroImage, title }) => {
  const classes = useStyles()
  return (
    <Card>
      <CardMedia>{heroImage}</CardMedia>
      <CardContent>
        <H2 classes={{ h2: classes.h2 }} gutterBottom={true}>
          {title}
        </H2>
        <GatsbyLinkButton
          color={"primary"}
          to={"/quero-aprender"}
          variant={"contained"}
        >
          Quero aprender
        </GatsbyLinkButton>
        &nbsp;
        <GatsbyLinkButton
          color={"initial "}
          to="/tire-suas-duvidas"
          variant={"contained"}
        >
          Tire suas d√∫vidas
        </GatsbyLinkButton>
      </CardContent>
    </Card>
  )
}

export default Hero
