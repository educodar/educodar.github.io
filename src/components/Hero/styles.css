.hero {
  padding: 20px;
  background-color: var(--hero-background-color);
  color: var(--hero-text-color);
}

.hero h1 {
  margin-bottom: 20px;
  font-weight: bold;
}

.hero .text {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
}
