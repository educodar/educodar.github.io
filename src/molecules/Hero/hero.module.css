.hero {
  composes: container from global;
  background-color: var(--background-color);
  color: var(--primary-color);
}

.hero h1 {
  font-weight: bold;
}

.hero .container {
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
}
