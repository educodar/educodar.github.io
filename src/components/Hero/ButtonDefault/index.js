import React from "react"
import "./styles.css"

const Button = ({ buttonText }) => (
  <button className="btn-default">{buttonText}</button>
)

export default Button
