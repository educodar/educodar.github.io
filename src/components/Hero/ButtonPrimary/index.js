import React from "react"
import "./styles.css"

const Button = ({ buttonText }) => (
  <button className="btn-primary">{buttonText}</button>
)

export default Button
