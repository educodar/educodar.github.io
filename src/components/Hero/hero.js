import React from "react"
import Button from "../Button/button"
import "./styles.css"

const Hero = () => (
  <div className="hero">
    <h1>Aulas de programação grátis para jovens</h1>
    <p>
      <Button buttonText="Button 1" />
      <Button buttonText="Button 2" />
    </p>
  </div>
)

export default Hero
