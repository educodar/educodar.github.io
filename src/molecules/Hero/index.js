import ComputerImage from "../../images/computer-line@2x.png"
import HeroImage from "../../atoms/HeroImage"
import React from "react"
import styles from "./hero.module.css"
import VerticalSpacing from "../../atoms/VerticalSpacing"
import ButtonLink from "../../atoms/ButtonLink"

const Hero = props => (
  <section>
    <HeroImage></HeroImage>
    <div className={styles.hero}>
      <div className={styles.container}>
        <h1>{props.title}</h1>
        <img src={ComputerImage} alt="computer"></img>
      </div>
      <VerticalSpacing size={"1x"}></VerticalSpacing>
      <div>
        <ButtonLink to="/quero-aprender" variant="primary">
          Quero aprender
        </ButtonLink>

        {/*<ButtonLink to="/tire-suas-duvidas" variant="primary" outline={true}>
          Tire suas d√∫vidas
</ButtonLink>*/}
      </div>
    </div>
    <VerticalSpacing size="4x" />
  </section>
)

export default Hero
