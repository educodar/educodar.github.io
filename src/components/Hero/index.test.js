import React from "react"
import renderer from "react-test-renderer"
import Hero from "./index"

describe("<Hero /> component", () => {
  /**
   * @todo fix failing test Hero(...): Nothing was returned from render
   */
  xit("renders correctly", () => {
    const tree = renderer.create(<Hero />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
