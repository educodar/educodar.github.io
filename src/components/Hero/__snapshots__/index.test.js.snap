// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> component renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root"
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 makeStyles-h2-1 MuiTypography-gutterBottom"
    />
    <a
      aria-disabled={false}
      className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiTypography-colorPrimary"
      href="/quero-aprender"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
      target="_self"
    >
      <span
        className="MuiButton-label"
      >
        Quero aprender
      </span>
    </a>
     
    <a
      aria-disabled={false}
      className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiButtonBase-root MuiButton-root MuiButton-contained MuiTypography-colorPrimary"
      href="/tire-suas-duvidas"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
      target="_self"
    >
      <span
        className="MuiButton-label"
      >
        Tire suas dúvidas
      </span>
    </a>
  </div>
</div>
`;
